File: microsoft/4/4.py
def merge(ar1, ar2, m, n):
    for i in range(n - 1, -1, -1):
        last = ar1[m - 1]
        j = m - 2
        while (j >= 0 and ar1[j] > ar2[i]):
            ar1[j + 1] = ar1[j]
            j -= 1
        if (j != m - 2 or last > ar2[i]):
            ar1[j + 1] = ar2[i]
            ar2[i] = last

ar1 = []
ar2 = []
n = int(input());
ar1 = [int(x) for x in input().split()]
m = int(input());
ar2 = [int(x) for x in input().split()]
m = len(ar1)
n = len(ar2)
merge(ar1, ar2, m, n)
for i in range(m):
    print(ar1[i], " ", end="")
for i in range(n):
    print(ar2[i], " ", end="")

File: Microsoft/4/4.cpp
#include <bits/stdc++.h>
using namespace std;

void merge(int ar1[], int ar2[], int m, int n) {
  for (int i = n - 1; i >= 0; i--) {
    int j, last = ar1[m - 1];
    for (j = m - 2; j >= 0 && ar1[j] > ar2[i]; j--)
      ar1[j + 1] = ar1[j];
    if (j != m - 2 || last > ar2[i]) {
      ar1[j + 1] = ar2[i];
      ar2[i] = last;
    }
  }
}

int main(void) {
  int n;
  cin >> n;
  int a[n];
  for (int i = 0; i < n; i++) {
    cin >> a[i];
  }
  int m;
  cin >> m;
  int b[m];
  for (int i = 0; i < m; i++) {
    cin >> b[i];
  }

  merge(a, b, n, m);

  for (int i = 0; i < n; i++)
    cout << a[i] << " ";
  for (int i = 0; i < m; i++)
    cout << b[i] << " ";
  return 0;
}
